sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -174,11 +191,14 @@ def __init__(self,
     self._state_handler_factory = GrpcStateHandlerFactory(
         self._state_cache, credentials)
     self._profiler_factory = profiler_factory
-    self._fns = KeyedDefaultDict(
-        lambda id: self._control_stub.GetProcessBundleDescriptor(
-            beam_fn_api_pb2.GetProcessBundleDescriptorRequest(
-                process_bundle_descriptor_id=id))
-    )  # type: Mapping[str, beam_fn_api_pb2.ProcessBundleDescriptor]
+
+    def default_factory(id):
+      # type: (str) -> beam_fn_api_pb2.ProcessBundleDescriptor
+      return self._control_stub.GetProcessBundleDescriptor(
+          beam_fn_api_pb2.GetProcessBundleDescriptorRequest(
+              process_bundle_descriptor_id=id))
+
+    self._fns = KeyedDefaultDict(default_factory)
```

Comment:
changing this to a function lets us add annotations which silences a mypy error.  mypy can now detect the key/value types of `KeyedDefaultDict`